### PR TITLE
Made Website into PWA

### PR DESCRIPTION
**Pull request by AdvaitDhingra**
https://alokprateek.in/articles/jekyll-pwa/ was used as a guide

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
I don't see an "Install" menu entry for https://root.cern/626 which I see it for other web apps?

**Comment by AdvaitDhingra** [comment]
> I don't see an "Install" menu entry for https://root.cern/626 which I see it for other web apps?

Yeah, for some reason it's not working anymore. I'll debug it and let you know

**Comment by AdvaitDhingra** [comment]
I think there was an issue with the way the relative url was written. I have tried to fix it. Crossed fingers it works.

File: _config.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ title_separator          : "-"
 subtitle                 : "Data Analysis Framework"
 name                     : "ROOT"
 description              : "Data Analysis Framework"
-url                      : "https://root.cern"
+url                      : "https://root.cern/"
```
Comment (Axel-Naumann) [inline _config.yml:24]: Please revert, this should not be needed

File: _config.yml [hunk 2]
```diff
@@ -306,3 +306,4 @@ feed:
   collections:
     - releases
     - ideas
+
```
Comment (Axel-Naumann) [inline _config.yml:309]: Please revert, this should not be needed

File: _includes/head/custom.html [hunk 3]
```diff
@@ -3,3 +3,22 @@
 <link rel="icon" type="image/png" href={{ "/assets/images/icon-32.png" | relative_url }} sizes="32x32">
 <link rel="icon" type="image/png" href={{ "/assets/images/icon-96.png" | relative_url }} sizes="96x96">
 <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300&amp;display=swap&amp;text=ROOTDataAnlysisFramework" rel="stylesheet">
+<head>
+  <link rel="manifest" href={{ "/manifest.json " | relative_url }}/>
```
Comment (Axel-Naumann) [inline _includes/head/custom.html:7]:
```suggestion
  <link rel="manifest" href={{ "/manifest.json" | relative_url }}/>
```

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]
Please fix the issue below and this should work!

**Comment by AdvaitDhingra** [comment]
> Please fix the issue below and this should work!

Okay. I initially placed the space there with the hope that it wouldnt add a / at the end. It seems to be causing problems. Also the serviceWorker isnt working properly. Quite weird since it works when built locally 😅

File: manifest.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "ROOT: analyzing petabytes of data, scientifically.",
+  "short_name": "ROOT",
+  "description": "ROOT Website",
+  "theme_color": "#346295",
+  "background_color": "#346295",
+  "display": "standalone",
+  "orientation": "portrait",
```
Comment (Axel-Naumann) [inline manifest.json:8]: I think landscape would be equally fine, no? Can we just leave this out, to allow both orientations?

Comment (AdvaitDhingra) [reply manifest.json:8]: Yes, we can do that. I just saw it on the guide I was using and assumed it was mandatory 😅

File: _includes/head/custom.html [hunk 5]
```diff
@@ -3,3 +3,22 @@
...
+<head>
+  <link rel="manifest" href={{ "/manifest.json" | relative_url }}/>
+  <script>
+    if ("serviceWorker" in navigator) {
+  if (navigator.serviceWorker.controller) {
+    console.log("An active service worker found, no need to register");
+  } else {
+    // Register the service worker
+    navigator.serviceWorker
+      .register("/serviceWorker.js", {
```
Comment (Axel-Naumann) [inline _includes/head/custom.html:15]: According to the console log, that URL needs to be ` | relative_url`-ified. Probably just
```suggestion
      .register( {{ "/serviceWorker.js" | relative_url }}, {
```

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: _includes/head/custom.html (hunk 5 above)
Comment (AdvaitDhingra) [reply _includes/head/custom.html:15]: Ah, good point.

File: manifest.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "scope": "/",
+  "start_url": "/",
+  "icons": [
+    {
+      "src": "/assets/images/icon180.png",
```
Comment (Axel-Naumann) [inline manifest.json:13]: This one would benefit from `| relative_url`, too, but I don't know how to register a `.json` file to be processed with liquid / by jekyll. Why don't you just try
```suggestion
      "src": {{"/assets/images/icon180.png" | relative_url}},
```

File: _includes/head/custom.html [hunk 7]
```diff
@@ -3,3 +3,22 @@
...
+<head>
+  <link rel="manifest" href={{ "/manifest.json" | relative_url }}/>
+  <script>
+    if ("serviceWorker" in navigator) {
+  if (navigator.serviceWorker.controller) {
+    console.log("An active service worker found, no need to register");
+  } else {
+    // Register the service worker
+    navigator.serviceWorker
+      .register( {{ "/serviceWorker.js" | relative_url }}, {
```
Comment (Axel-Naumann) [inline _includes/head/custom.html:15]: Almost! This gets transformed into `.register( /626/serviceWorker.js, {` - so we probably need:
```suggestion
      .register( '{{ "/serviceWorker.js" | relative_url }}', {
```

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: _includes/head/custom.html [hunk 8]
```diff
@@ -3,3 +3,22 @@
...
+  <link rel="manifest" href={{ "/manifest.json" | relative_url }}/>
+  <script>
+    if ("serviceWorker" in navigator) {
+  if (navigator.serviceWorker.controller) {
+    console.log("An active service worker found, no need to register");
+  } else {
+    // Register the service worker
+    navigator.serviceWorker
+      .register( '{{ "/serviceWorker.js" | relative_url }}', {
+        scope: "/"
```
Comment (Axel-Naumann) [inline _includes/head/custom.html:16]: Chromium:
```
The path of the provided scope ('/') is not under the max scope allowed ('/626/'). Adjust the scope, move the Service Worker script, or use the Service-Worker-Allowed HTTP header to allow the scope.
```
```suggestion
        scope: '{{ "/" | relative_url }}'
```

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: _includes/head/custom.html (hunk 8 above)
Comment (AdvaitDhingra) [reply _includes/head/custom.html]: Ahh. I didn't know a solution for this scope problem.

**Comment by AdvaitDhingra** [comment]
Sorry, I was gone for a bit.

File: _includes/head/custom.html [hunk 9]
```diff
@@ -3,3 +3,22 @@
 <link rel="icon" type="image/png" href={{ "/assets/images/icon-32.png" | relative_url }} sizes="32x32">
 <link rel="icon" type="image/png" href={{ "/assets/images/icon-96.png" | relative_url }} sizes="96x96">
 <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300&amp;display=swap&amp;text=ROOTDataAnlysisFramework" rel="stylesheet">
+<head>
+  <link rel="manifest" href={{ "/manifest.json" | relative_url }}/>
```
Comment (Axel-Naumann) [inline _includes/head/custom.html:7]: That gives `<link rel="manifest" href=/626/manifest.json/>` so we need
```suggestion
  <link rel="manifest" href="{{ "/manifest.json" | relative_url }}" />
```

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

**Comment by AdvaitDhingra** [comment]
The serviceWorker is working 🚀

**Review by Axel-Naumann — APPROVED** [review]

**Pull request merged**